CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -796,27 +815,27 @@ private static function updateCacheFromTempTable(CRM_Utils_SQL_TempTable $groupC
    */
   protected static function insertGroupContactsIntoTempTable(string $tempTableName, int $groupID, ?int $savedSearchID, ?string $children): void {
     if ($savedSearchID) {
-      $ssParams = CRM_Contact_BAO_SavedSearch::getSearchParams($savedSearchID);
+      $savedSearch = SavedSearch::get(FALSE)
+        ->addWhere('id', '=', $savedSearchID)
+        ->addSelect('*')
+        ->execute()
+        ->first();
 
       $excludeClause = "NOT IN (
                         SELECT contact_id FROM civicrm_group_contact
                         WHERE civicrm_group_contact.status = 'Removed'
                         AND civicrm_group_contact.group_id = $groupID )";
       $addSelect = "$groupID AS group_id";
 
-      if (!empty($ssParams['api_entity'])) {
-        $sql = self::getApiSQL($ssParams, $addSelect, $excludeClause);
+      if ($savedSearch['api_entity']) {
+        $sql = self::getApiSQL($savedSearch, $addSelect, $excludeClause);
       }
       else {
-        // CRM-7021 rectify params to what proximity search expects if there is a value for prox_distance
-        if (!empty($ssParams)) {
-          CRM_Contact_BAO_ProximityQuery::fixInputParams($ssParams);
-        }
-        if (isset($ssParams['customSearchID'])) {
-          $sql = self::getCustomSearchSQL($savedSearchID, $ssParams, $addSelect, $excludeClause);
+        if (!empty($savedSearch['form_values']['customSearchID'])) {
```

Comment:
@eileenmcnaughton is this right why is it buried in form_values?

Reply:
Yeah - why is it! But, it is. that form_values field saved to the db is what holds it